secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,235 @@
+
+	// forceSynchronization can be used to force the secret synchronization. The secret synchronization is
+	// triggered, by changing the value in this field.
+	// This field is not used to resolve synchronization conflicts.
+	// It is not related with the force query parameter in the Apply operation.
+	// https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=^[A-Za-z0-9]([-A-Za-z0-9]+([-._a-zA-Z0-9]?[A-Za-z0-9])*)?
+	// +optional
+	ForceSynchronization string `json:"forceSynchronization,omitempty"`
+}
+
+// SecretSyncStatus defines the observed state of the secret synchronization process.
+type SecretSyncStatus struct {
+	// syncHash contains the hash of the secret object data, data from the SecretProviderClass (e.g. UID,
+	// and metadata.generation), and similar data from the SecretSync. This hash is used to
+	// determine if the secret changed.
+	// The hash is calculated using the HMAC (Hash-based Message Authentication Code) algorithm, using bcrypt
+	// hashing, with the SecretsSync's UID as the key.
+	// The secret is updated if:
+	//		1. the hash is different
+	//		2. the lastRetrievedTimestamp indicates a rotation is required
+	//			- the rotation poll interval is passed as a parameter in the controller configuration
+	//		3. the UpdateStatus is 'Failed'
+	// +optional
+	SyncHash string `json:"syncHash,omitempty"`
+
+	// lastSuccessfulSyncTime represents the last time the secret was retrieved from the Provider and updated.
+	// +optional
+	LastSuccessfulSyncTime *metav1.Time `json:"lastSuccessfulSyncTime,omitempty"`
+
+	// conditions represent the status of the secret create and update processes.
+	// The status is set to True if the secret was created or updated successfully.
+	// The status is set to False if the secret create or update failed and the reconciliation loop won't retry
+	// the operation until the an action is performed by the user.
+	// The status is set to Unknown if for example the secret patch failed due to an unknown error and
+	// the reconciliation loop will retry the operation.
+	// The following conditions are used:
+	//		- Type: CreateSucceeded
+	//			Status: True
+	//			Reason: CreateSucceeded
+	//			Message: The secret was created successfully.
+	//		- Type: CreateFailedProviderError
+	//			Status: Unknown
+	//			Reason: ProviderError
+	//			Message: The secret create failed due to a provider error, check the logs or the events for more information.
+	//		- Type: CreateFailedInvalidLabel
+	//			Status: False
+	//			Reason: InvalidClusterSecretLabelError
+	//			Message: The secret create failed because a label reserved for the controller is applied on the secret.
+	//		- Type: CreateFailedInvalidAnnotation
+	//			Status: False
+	//			Reason: InvalidClusterSecretAnnotationError
+	//		    Message: The secret create failed because an annotation reserved for the controller is applied on the secret.
```

Comment:
I think the CreateSucceeded/CreateFailed___ conditions could all collapse to a single SecretCreated condition

* On successful create: `type: SecretCreated, status: True`
* On failed create, one of `type: SecretCreated, status: False, reason: ProviderError | InvalidClusterSecretLabelError InvalidClusterSecretAnnotationError | ...`

Reply:
similar pattern for recording health of the most recent update attempt

a single SecretUpdated condition, which is true if the last update attempt succeeded, and false if the last update attempt failed, with reason set to the particular reason for the failure

reasons could be:
* granular reasons about failures fetching the data from the provider at various stages
* coarse-grained reasons about the server rejecting the update (updateforbidden, updateinvalid, updatebadrequest)
  * if we know the message from the server is safe to surface to the end user (maaaaybe if the error that comes back has specific field path errors / details, or if we recognize our own validating admission policy messages), can include that verbatim or truncated in the condition message
  * otherwise, put a generic condition message